cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -611,16 +612,27 @@ func (p *BicepProvider) Deploy(ctx context.Context) (*provisioning.DeployResult,
 		return nil, err
 	}
 
-	err = p.validatePreflight(
-		ctx,
-		bicepDeploymentData.Target,
-		bicepDeploymentData.CompiledBicep.RawArmTemplate,
-		bicepDeploymentData.CompiledBicep.Parameters,
-		deploymentTags,
-		optionsMap,
-	)
-	if err != nil {
-		return nil, err
+	if strings.ToLower(p.env.Getenv(environment.DisablePreflightName)) != "true" {
```

Comment:
`.Getenv()` falls back to check the system environment (not only the values from azd's .env file)

Do we want to make it only for azd's .env file?

In the `warningMessage` below we are telling folks they can run `azd env set XXXXX false` or remove the value from the .env;  If the env var is set on system environment, should we re-word the error?